Refit/RequestBuilderImplementation.cs

```diff
@@ -404,16 +404,15 @@ private async Task<T> DeserializeContentAsync<T>(HttpResponseMessage resp, HttpC
         {
             var kvps = new List<KeyValuePair<string, object>>();
 
-            var props = dictionary.Keys;
-            foreach (string key in props)
+            foreach (var key in dictionary.Keys)
             {
                 var obj = dictionary[key];
                 if (obj == null)
                     continue;
 
                 if (DoNotConvertToQueryMap(obj))
                 {
-                    kvps.Add(new KeyValuePair<string, object>(key, obj));
+                    kvps.Add(new KeyValuePair<string, object>(key.ToString(), obj));
```

Comment:
In general this looks good -- but we should not call ToString() as that makes it harder to override the value for types you don't control. Instead it should probably delegate to the `settings.UrlParameterFormatter` to do the format.

https://github.com/reactiveui/refit/blob/master/Refit/RefitSettings.cs#L24

Reply:
Thanks for the helpful feedback! I'll dig into this and hopefully have changes to this PR out sometime tomorrow.

Reply:
@clairernovotny I've made updates to the PR to ensure that the query map respects provided formatters. My unit tests match my assumptions on what the generated query string is supposed to look like with the formatters, but please let me know if I'm missing something. Thanks again!